### ISSUE TITLE
defaultTo() doesn't JSON.stringify jsonb columns

### ISSUE THREAD

**Issue by elazar** [tgriesser/knex#3167]
# Environment

Knex version: 0.11.0, but this likely applies to the latest release
Database + version: PostgreSQL 11.2
OS: OS X 10.14.4

# Bug

I think the fix for #280 in bd3147964cf6fb6693fe73e991494448e54721b4 should have been applied to the `jsonb` type as well as the `json` type on [this line](https://github.com/tgriesser/knex/blob/0e845489ca742f4052bbe4f717e81c24bb1544f5/src/schema/columncompiler.js#L157).

Here's an example migration:

```js
exports.up = function(knex, Promise) {
     return Promise.all([
         knex.schema.table("users", function (table) {
            table.jsonb("subscriptions").defaultTo({"marketing_subscribed": true});
         })
     ]);
 };
```

This results in the following error:

```
Knex:warning - migrations failed with error: alter table "users" add column "subscriptions" jsonb default '[object Object]' - invalid input syntax for type json
error: invalid input syntax for type json
    at Connection.parseE (/Users/matt/Code/api-users/node_modules/pg/lib/connection.js:567:11)
    at Connection.parseMessage (/Users/matt/Code/api-users/node_modules/pg/lib/connection.js:391:17)
    at Socket.<anonymous> (/Users/matt/Code/api-users/node_modules/pg/lib/connection.js:129:22)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:597:20)
```

Applying `JSON.stringify()` to the value passed to `defaulTo()` as shown below appears to works around the issue:

```
table.jsonb("subscriptions").defaultTo(JSON.stringify({"marketing_subscribed": true}));
```

This produces the following field when running `\d` against the table from the `psql` client:

```
          Column            |           Type           | Collation | Nullable |                 Default
-----------------------------+--------------------------+-----------+----------+-----------------------------------------
 subscriptions               | jsonb                    |           |          | '{"marketing_subscribed": true}'::jsonb
```

**Comment by lorefnon** [comment]
I don't think the original fix (https://github.com/tgriesser/knex/commit/bd3147964cf6fb6693fe73e991494448e54721b4) actually worked (Ref: https://github.com/tgriesser/knex/commit/bd3147964cf6fb6693fe73e991494448e54721b4#r6637388)

**Comment by lorefnon** [comment]
This was fixed in https://github.com/tgriesser/knex/pull/3318 and is available in [0.18.2](https://www.npmjs.com/package/knex/v/0.18.2).

**Comment by billybonks** [comment]
Can close the issue if it's fixed ?
